tsconfig.json

```diff
@@ -12,13 +12,13 @@
     "checkJs": false /* Enable error reporting in type-checked JavaScript files. */,
     "declaration": true,
     "emitDeclarationOnly": true,
-    "declarationDir": "dist/dts",
+    "declarationDir": "build/types",
```

Comment:
Why not just `build`?

Reply:
Also, maybe we can just use `rootDir: 'src'` and `outDir: 'build'`?

Reply:
It's not a big deal, but I guess for clarity in case we want to plonk other temporary files in here in future?

Reply:
And re `rootDir: 'src'` and `outDir: 'build'`, I guess we could: do you think there's some advantage to that?

Reply:
> but I guess `build/types` for clarity

I think it can be nice that `index.(m)js` can have a sibling `index.d.(m)ts`. But yeah it doesn't serve a real purpose, because 1. we bundle anyway and 2. we route to types via `package.json` field anyway, not via some sibling default type resolution.

> do you think there's some advantage to that?

nothing except reducing depth in `build/types/src/...` vs `build/types/...`